fix(k8s): fix builtdns annotations (#4660)
kumahq/kuma: UPGRADE.md

### Helm

- Under `cni.image`, the default values for `repository` and `registry` have been
changed to agree with the other `image` values.

> - We are deprecating `kuma.io/builtindns` and `kuma.io/builtindnsport` annotations in favour or the clearer `kuma.io/builtin-dns` and `kuma.io/builtin-dns-port` the code is backward compatible

Reviewer: > the code is backward compatible

What does this mean? That no action is required but users _should_ migrate?

Author: Yes I guess this wording is wrong. Do you have an idea on how to rephrase?

> - We are deprecating `kuma.io/builtindns` and `kuma.io/builtindnsport` annotations in favour or the clearer `kuma.io/builtin-dns` and `kuma.io/builtin-dns-port` the code is backward compatible
> but you should migrate (a warning is present on the log if you are using the deprecated version).

Reviewer: ```suggestion
- We are deprecating `kuma.io/builtindns` and `kuma.io/builtindnsport` annotations in favour of the clearer `kuma.io/builtin-dns` and `kuma.io/builtin-dns-port`. The behavior of the new annotations is unchanged but you should migrate (a warning is present on the log if you are using the deprecated version).
```

Like that maybe?
